### ISSUE TITLE
Delete faulty almaMmsId mappings that are refering to local records

### ISSUE THREAD

**Issue by TobiasNx** [hbz/rpb#107]
conf/maps/almaMmsId-to-rpbId.tsv has some mappings that are oviously wrong.
They are not mapped to a NZ record these would have at least 6441 at the end of the almaMmsId.
But they are also mapping to completly wrong or non existing records.

Not sure how they were mapped:

6452
991000117199706452	107t01165452
991000087399706452	107t01165454
6449
991010785389706449	107t01165591
6456
991005467559706456	929t01077759

**Comment by TobiasNx** [comment]
Also there are a lot of duplicate almaMmsId in the file:
e.g.

```
990218653550206441	929t172910
990218653550206441	929t172911
990218653550206441	929t172912
990218653550206441	929t172913
990218653550206441	929t172914
990218653550206441	929t172915
990218653550206441	929t172916
990218653550206441	929t172917
990218653550206441	929t172918
990218653550206441	929t172919
990218653550206441	929t172920
990218653550206441	929t172921
990218653550206441	929t172922
990218653550206441	929t172923
990218653550206441	929t172924
990218653550206441	929t172925
990218653550206441	929t172926
990218653550206441	929t172927
990218653550206441	929t172928
990218653550206441	929t172929
990218653550206441	929t172930
990218653550206441	929t172931
990218653550206441	929t172932
990218653550206441	929t172933
990218653550206441	929t172934
990218653550206441	929t172935
990218653550206441	929t172936
```

This seems to be related to https://jira.hbz-nrw.de/projects/RPB/issues/RPB-232

**Comment by TobiasNx** [comment]
Does it make sense to recreate this mapping file after fixing [jira.hbz-nrw.de/projects/RPB/issues/RPB-232](https://jira.hbz-nrw.de/projects/RPB/issues/RPB-232)

**Comment by acka47** [comment]
> Does it make sense to recreate this mapping file after fixing [jira.hbz-nrw.de/projects/RPB/issues/RPB-232](https://jira.hbz-nrw.de/projects/RPB/issues/RPB-232)

Sounds reasonable. +1

**Comment by fsteeg** [comment]
As I commented in https://github.com/hbz/rpb/pull/108, I don't understand. As we discussed on the RPB channel recently, isn't this only the original file created here, which has then been moved to lookup-tables?

https://github.com/hbz/rpb/commits/main/conf/maps/almaMmsId-to-rpbId.tsv
https://github.com/hbz/lookup-tables/commits/master/data/almaMmsId2rpbId.tsv

**Comment by TobiasNx** [comment]
No the filesize is a lot bigger here in contrast to lookuptables. (+45.000) The file here seems to be based on all records that HAVE an hbzId in the strapi data. While the one in lookup tables (size ~10000) is based on the reconciliation done by the RPB Team.

**Comment by acka47** [comment]
Wir können/sollten im rpb-Repo auch einfach Deutsch schreiben...

Bleibt die Frage: Wird diese Datei momentan überhaupt für irgendetwas verwendet? Wenn nicht, können und sollten wir das auf später schieben. Wie im gestrigen Treffen gesagt, müssen wir eines nach dem anderen machen. Der momentane Fokus ist auf RPB-184 und damit hat das hier nichts zu tun und verschafft uns nur zusätzliche Mental Load.

**Comment by acka47** [comment]
Wir haben heute im RPB-Treffen besprochen, dass wir die aus RPB-50 entstandene Alma-MMS-ID-zu-RPB-ID-Konkordanz nutzen, um bei den enstprechenden Strapi-Titeln im Zuge eines Reimports das `hbzID`-Feld auszufüllen, siehe https://jira.hbz-nrw.de/browse/RPB-184?focusedId=330426&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-330426

Dafür sollten wir dann wohl die Liste unter https://github.com/hbz/lookup-tables/blob/master/data/almaMmsId2rpbId.tsv verwenden. Dort gibt es auch nur zwei Titel, die nicht in der NZ sind:

```
$ curl https://raw.githubusercontent.com/hbz/lookup-tables/refs/heads/master/data/almaMmsId2rpbId.tsv | grep -v 6441 
0almaMmsId	rpbId
990000167930306443	RPB929t062907
#983: HT018569020	RPB929t01074097
```

**Comment by acka47** [comment]
To Do: 

- [ ] die beiden Nicht-NZ-Titel aus dem vorherigen Kommentar korrigieren
- [ ] Nicht-NZ-Titel aus de rLookup-Tabelle entfernen, siehe https://github.com/hbz/lookup-tables/issues/61

**Comment by TobiasNx** [comment]
Workflow to add HT numbers: https://github.com/TobiasNx/metafacture_workflows/tree/master/getHbzIdFromLobid

**Comment by TobiasNx** [comment]
Ich hab den Metafacture Workflow ([TobiasNx/metafacture_workflows@master/getHbzIdFromLobid](https://github.com/TobiasNx/metafacture_workflows/tree/master/getHbzIdFromLobid?rgh-link-date=2025-02-04T13%3A54%3A37.000Z) soweit angepasst, dass er alle TODOs erfüllen würde:


> To Do:
> 
>     * [ ]  die beiden Nicht-NZ-Titel aus dem vorherigen Kommentar korrigieren[ ]  Nicht-NZ-Titel aus de rLookup-Tabelle entfernen, siehe [Update rpb lookuptables lookup-tables#61](https://github.com/hbz/lookup-tables/issues/61)[ ]  Spalte mit `hbzId` ergänzen

Wir müssten uns die Ausreißer, die in den logs auftauchen mal einzelnd anschauen undhändisch verbessern und anschauen, warum einige resourcen KEINE rpbId haben, obwohl sie in der Ausgangstabelle sind.

@fsteeg wäre die Datei: https://github.com/TobiasNx/metafacture_workflows/blob/master/getHbzIdFromLobid/rpbLobidMappingWithHT.tsv für eine Anforderungen gut genug?

**Comment by fsteeg** [comment]
> @fsteeg wäre die Datei: https://github.com/TobiasNx/metafacture_workflows/blob/master/getHbzIdFromLobid/rpbLobidMappingWithHT.tsv für eine Anforderungen gut genug?

Für das Beispiel aus RPB-184, nämlich https://rpb.lobid.org/929t982199, sieht das gut aus, für die rpbId 929t982199 kann ich da die hbzId HT009968937 finden. Davon abgesehen kann ich das nicht gut beurteilen. Vielleicht kann @acka47 auch mal drauf schauen. Und wenn es passt, sollte es vermutlich ein PR gegen https://github.com/hbz/lookup-tables/blob/master/data/almaMmsId2rpbId.tsv  sein, oder?

**Comment by acka47** [comment]
> Vielleicht kann @acka47 auch mal drauf schauen. Und wenn es passt, sollte es vermutlich ein PR gegen https://github.com/hbz/lookup-tables/blob/master/data/almaMmsId2rpbId.tsv sein, oder?

Ja, mach bitte einen PR in lookup-tables, @TobiasNx , und weise ihn mir zum Review zu,

**Comment by acka47** [comment]
> Ja, mach bitte einen PR in lookup-tables, @TobiasNx , und weise ihn mir zum Review zu,

Hast du schon getan, sehe ich gerade: https://github.com/hbz/lookup-tables/pull/62#event-16183883506 Danke! Und sorry for the noise...
